pkg/auth/dependency/oauth/handler/handler_token.go

```diff
@@ -204,6 +204,10 @@ func (h *TokenHandler) handleRefreshToken(
 		return nil, err
 	}
 
+	if h.Time.NowUTC().After(offlineGrant.ExpireAt) {
```

Comment:
In normal case, shouldn't the offline grant be evicted from redis? We do set expiry on it. Anyway it is good to have checking here. The underlying storage may not be redis.

Reply:
Yes indeed, just a double check.